### ISSUE TITLE
Bright background

### ISSUE THREAD

**Issue by mrev11** [xtermjs/xterm.js#4459]
Look at these escape sequences:

\033[2J
\033[0;1;7;31;47m
\033[1;0H abcdefgh
\033[0;1;37;41m
\033[2;0H abcdefgh
\033[0m

line 1: clear screen
line 2: clear;bright;reverse;fg=red;bg=white
line 3: goto(1,0) and write
line 4: clear;bright;fg=white;bg=red
line 5: goto(2,0) and write

Echoing these sequences we get the following results:

xterm
    first line: white fg on bright red bg
    second line: bright white fg on red bg

Xfce4 terminal (the same as xterm)
    first line: white fg on bright red bg
    second line: bright white fg on red bg

xtermjs (first and second line are the same)
    first line: bright white fg on red bg
    second line: bright white fg on red bg

Conclusion: xtermjs cannot draw bright bg.

Screenshot of xterm:

![Screenshot from 2023-03-30 12-42-51](https://user-images.githubusercontent.com/23376284/228821536-03b248f4-ab19-4f72-a033-5027a7096fa5.png)

Screenshot of Xfce4-terminal

![Screenshot from 2023-03-30 12-43-56](https://user-images.githubusercontent.com/23376284/228821562-fe14a1e5-2b07-44ce-ada2-22dcf77e6ff7.png)

Screenshot of Chromium
![Screenshot from 2023-03-30 12-54-04](https://user-images.githubusercontent.com/23376284/228821585-3e13bf58-af39-49e6-bb56-7c889c7b061d.png)

**Comment by jerch** [comment]
Eww thats really a tricky one. I second @Tyriar - applying the bold|bright FG setting to BG here seems weird.

The problem with SGR 1 is, that it is not really clear, how it should influence font weight, colors or both. Some say "nope that means bright, because old terminals has no notion of bold glyphs", others say "thats not true, in fact its was a mixture of both back then, because of you give more power on the beam, the glyph will shine brighter _and_ bold". Imho a bold font variant is the right way to go on these days TEs, as we have explicit bright colors as well, theoretically giving us 4 mixture possibilities (normal + normal, bold + normal, normal + bright, bold + bright).

To me it seems the other TEs just apply it to the colors anyway, and then flip the colors on SGR 7. I think we eval the bold setting very late at rendering?

**Comment by Tyriar** [comment]
Yeah the more modern approach is to disable `drawBoldTextInBrightColors` anyway, so this isn't a big deal. We didn't change the default as it would probably result in a bunch of issue reports.

**Comment by mrev11** [comment]
Weird trick. It is the standard technique by which libcurses based programs use 16 bg colors. 
Terminal interprets the escape sequences emitted by  curses programs. What are the possibilities?

0x1b[40m  black
...
0x1b[47m  white

Only 8 colors. But additionally there is the bold/bright attribute. If bright attribute would be applied always to the fg, curses programs couldn't  use no more than 8 bg colors. For this reason, if bright and reverse attribute are both present, terminals draw bright bg. This feature is supported by  xterm and Xfce4 (vte) terminal.

**Comment by Tyriar** [comment]
@mrev11 I say the more modern approach is to disable `drawBoldTextInBrightColors` because you can explicitly apply bright colors separate from bold:

```
          If 16-color support is compiled, the following aixterm
          controls apply.  Assume that xterm's resources are set so that
          the ISO color codes are the first 8 of a set of 16.  Then the
          aixterm colors are the bright versions of the ISO colors:

            Ps = 9 0  ⇒  Set foreground color to Black.
            Ps = 9 1  ⇒  Set foreground color to Red.
            Ps = 9 2  ⇒  Set foreground color to Green.
            Ps = 9 3  ⇒  Set foreground color to Yellow.
            Ps = 9 4  ⇒  Set foreground color to Blue.
            Ps = 9 5  ⇒  Set foreground color to Magenta.
            Ps = 9 6  ⇒  Set foreground color to Cyan.
            Ps = 9 7  ⇒  Set foreground color to White.
            Ps = 1 0 0  ⇒  Set background color to Black.
            Ps = 1 0 1  ⇒  Set background color to Red.
            Ps = 1 0 2  ⇒  Set background color to Green.
            Ps = 1 0 3  ⇒  Set background color to Yellow.
            Ps = 1 0 4  ⇒  Set background color to Blue.
            Ps = 1 0 5  ⇒  Set background color to Magenta.
            Ps = 1 0 6  ⇒  Set background color to Cyan.
            Ps = 1 0 7  ⇒  Set background color to White.
```

https://invisible-island.net/xterm/ctlseqs/ctlseqs.html

**Comment by mrev11** [comment]
@Tyriar  Thank you for the info, I will study it.
